src/shogun/clustering/KMeansBase.cpp

```diff
@@ -354,17 +355,23 @@ SGMatrix<float64_t> CKMeansBase::kmeanspp()
 
 void CKMeansBase::init()
 {
-	max_iter=300;
-	k=8;
-	dimensions=0;
-	fixed_centers=false;
-	use_kmeanspp=false;
-	SG_ADD(&max_iter, "max_iter", "Maximum number of iterations", ParameterProperties::HYPER);
-	SG_ADD(&k, "k", "k, the number of clusters", ParameterProperties::HYPER);
+	max_iter = 300;
+	k = 8;
+	dimensions = 0;
+	fixed_centers = false;
+	use_kmeanspp = false;
+	SG_ADD(
+	    &max_iter, "max_iter", "Maximum number of iterations",
+	    ParameterProperties::HYPER);
+	SG_ADD_CONSTRAINT(
+	    &k, "k", "k, the number of clusters",
+	    ParameterProperties::HYPER | ParameterProperties::CONSTRAIN,
+	    positive<>());
```

Comment:
the `<>` can only be dropped in c++17. the template is by default double (it can't be inferred here)